extension-style/src/main/java/com/mapbox/maps/extension/style/layers/LayerExt.kt

```diff
@@ -40,14 +40,18 @@ fun StyleManagerInterface.getLayer(layerId: String): Layer? {
 }
 
 /**
- * Tries to cast the Layer to T, throws ClassCastException if it's another type.
+ * Tries to cast the Layer to T.
  *
  * @param layerId the layer id
  * @return T
  */
-fun <T : Layer> StyleManagerInterface.getLayerAs(layerId: String): T {
-  @Suppress("UNCHECKED_CAST")
-  return getLayer(layerId) as T
+inline fun <reified T : Layer> StyleManagerInterface.getLayerAs(layerId: String): T? {
+  val layer = getLayer(layerId)
```

Comment:
imho it will already crash here if layer is not `T` but I might be wrong.
perhaps:
```suggestion
  val layer = getLayer(layerId) as? T
  if (layer == null) {
    Logger.e("StyleLayerPlugin", "Given layerId = $layerId is not requested type in Layer")
    return null
  }
  return layer
```

Reply:
but maybe it won't work my way, I did not check it 😄

Reply:
Yeah, it will work as well.